gui/default/syncthing/core/syncthingController.js

```diff
@@ -549,10 +564,28 @@ angular.module('syncthing.core')
                 $scope.connectionsTotal = data.total;
 
                 data = data.connections;
+                var currentAddresses = [];
                 for (id in data) {
                     if (!data.hasOwnProperty(id)) {
                         continue;
                     }
+                    if (Object.keys(data).length != $scope.devices.length) {
+                        // Avoid errors when called before first updateLocalConfig()
+                        continue;
+                    }
+                    if (!(id in $scope.idToRemoteGUI)) {
+                        $scope.idToRemoteGUI[id] = "";
+                    }
+                    var port = $scope.findDevice(id).remoteGUIPort;
```

Comment:
`findDevice()` removed on main branch, as `$scope.devices` is now a map itself:

```suggestion
                    var port = $scope.devices[id].remoteGUIPort;
```